c++/CMakeLists.txt

```diff
@@ -51,6 +46,10 @@ if(CAPNP_LITE AND BUILD_TOOLS)
   message(WARNING "Command-line tools will not be built with CAPNP_LITE.")
 endif()
 
+if(BUILD_TOOLS AND EXTERNAL_CAPNP)
+  message(SEND_ERROR "EXTERNAL_CAPNP cannot be set with BUILD_TOOLS.")
```

Comment:
1. I'm not sure why I added `BUILD_TOOLS`, as simply running `make capnpc` has the same effect. I guess it's redundant.

2. `BUILD_TESTING` is defined by CTest and controls whether CMake generates test targets. Namespacing CMake variables is generally unnecessary as independent projects shouldn't be loaded together (the `ExternalProject` module should be used to build dependencies). When I tested this patch, it definitely seemed backwards compatible (source-wise with up-to-date CMake versions).